src/stores/disk.rs

```diff
@@ -139,6 +183,24 @@ where
                 }
             }
         }
+
+        if self.sync_to_disk_on_cache_change {
+            // NOTE: we are not returning any error here so as not to change the function signature,
+            // and break backwards compatibility
+            // TODO: Review changing the function signature to return a Result<(), DiskCacheError> - this would be a breaking change
```

Comment:
I'm fine with changing the signature to raise any errors here

Reply:
@jaemk thanks, I've updated this and the changelog, thanks for the initial review